src/main/webapp/app/shared/user-settings/ide-preferences/ide-settings.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Ide, IdeMappingDTO } from 'app/shared/user-settings/ide-preferences/ide.model';
+import { ProgrammingLanguage } from 'app/entities/programming-exercise.model';
+
+@Injectable({ providedIn: 'root' })
+export class IdeSettingsService {
+    public ideSettingsUrl = 'api/ide-settings';
+    error?: string;
+
+    programmingLanguageToIde: Map<ProgrammingLanguage, Ide>;
+
+    constructor(private http: HttpClient) {
+        this.loadIdePreferences();
+    }
+
+    /**
+     * GET call to the server to receive the stored ide preferences of the current user
+     * @return the saved ide preference which were found in the database or error
+     */
+    public loadIdePreferences(): void {
+        this.http.get<IdeMappingDTO[]>(this.ideSettingsUrl, { observe: 'response' }).subscribe((res) => {
+            this.programmingLanguageToIde = new Map(res.body?.map((x) => [x.programmingLanguage, x.ide]));
+            if (this.programmingLanguageToIde.size === 0) {
+                this.programmingLanguageToIde = new Map([[ProgrammingLanguage.EMPTY, PREDEFINED_IDE[0]]]);
+            }
+        });
+    }
+
+    /**
+     * PUT call to the server to update a stored ide preferences of the current user
+     * @return the newly saved ide preference or error
+     */
```

Comment:
I suspect that the client style will complain here because of missing `@param`s for programmingLanguage and ide. Can you add them? (other methods in this file as well)